Neos.Neos/Classes/Service/LinkingService.php

```diff
@@ -338,14 +338,17 @@ public function createSiteUri(ControllerContext $controllerContext, Site $site):
             throw new NeosException(sprintf('Cannot link to a site "%s" since it has no active domains.', $site->getName()), 1460443524);
         }
         $requestUri = $controllerContext->getRequest()->getHttpRequest()->getUri();
-        $baseUri = $controllerContext->getRequest()->getHttpRequest()->getBaseUri();
+
+        /** @var UriInterface $baseUri */
+        $baseUri = $controllerContext->getRequest()->getHttpRequest()->getAttribute(ServerRequestAttributes::BASE_URI);
+
         $port = $primaryDomain->getPort() ?: $requestUri->getPort();
         return sprintf(
             '%s://%s%s%s',
             $primaryDomain->getScheme() ?: $requestUri->getScheme(),
             $primaryDomain->getHostname(),
             $port && !in_array($port, [80, 443], true) ? ':' . $port : '',
-            rtrim($baseUri->getPath(), '/') // remove trailing slash, $uri has leading slash already
+            rtrim((string)$baseUri, '/') // remove trailing slash, $uri has leading slash already
```

Comment:
Wait, that is no longer only the path, is it?

Reply:
@daniellienert / @kitsunet What about this one? I did not get around to check in detail, because a ton of other things came up during the testing I did today so far. But that seems as if it _must_ be broken…

Reply:
yep, `$baseUri` at that stage would be the full URI including host, but what Christian said. The ServerRequestAttribute will vanish anyway and this needs adjustment then.

Reply:
I still resent merging something we agree is broken. I'll push a fix.

Reply:
I still resent merging something we agree is broken. I'll push a fix.